### PR TITLE
Support Python 3.5

### LINKED ISSUE #1491: Python 3.5 support
Luigi may already work with Python 3.5, but since the README doesn't mention it I thought I'd ask.

Does Luigi support Python 3.5?

### PR DESCRIPTION
**Pull request by nchammas**
Not sure if testing on Python 3.5 is as simple as the changes I made. If this works, I'll update the README and docs as well and clean the PR up.

Fixes #1491.

### CODE REVIEW

**Comment by nchammas** [comment]
It looks like Tox is [not finding Python 3.5](https://travis-ci.org/spotify/luigi/jobs/101147744#L163):

> ```
ERROR: InterpreterNotFound: python3.5
```

Did I miss a config somewhere?

Travis does support Python 3.5, as I mentioned in #1491, and here is an [example build from simplejson showing a Travis build on Python 3.5](https://travis-ci.org/simplejson/simplejson/jobs/92528889#L90).

**Comment by erikbern** [comment]
I think it's this problem: https://github.com/travis-ci/travis-ci/issues/4794

**Comment by nchammas** [comment]
Thanks for the tip. I'll try out the ["matrix trick"](https://github.com/travis-ci/travis-ci/issues/4794#issuecomment-143758799) described there.

**Comment by nchammas** [comment]
The `nonhdfs` build failed, but the `unixsocket` and `cdh` builds succeeded. 

[Here's the problem](https://travis-ci.org/spotify/luigi/jobs/101178502#L2982-L2996) with the `nonhdfs` build:

```
======================================================================
ERROR: test_raise_unpicklable_exception_in_complete (worker_parallel_scheduling_test.ParallelSchedulingTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/spotify/luigi/.tox/py35-nonhdfs/lib/python3.5/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/home/travis/build/spotify/luigi/test/worker_parallel_scheduling_test.py", line 152, in test_raise_unpicklable_exception_in_complete
    self.assertRaises(pickle.PicklingError, pickle.dumps, ex)
  File "/opt/python/3.5.0/lib/python3.5/unittest/case.py", line 724, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/opt/python/3.5.0/lib/python3.5/unittest/case.py", line 176, in handle
    callable_obj(*args, **kwargs)
AttributeError: Can't pickle local object 'UnpicklableExceptionTask.complete.<locals>.UnpicklableException'
```

I'm guessing the `AttributeError` is unexpected here. [Relevant doc](https://github.com/python/cpython/blob/0710cee1bb5865f0c9906f57dee5a28aac8b7e8d/Lib/pickle.py#L75-L84):

> ```
    Note that other exceptions may also be raised during unpickling, including
    (but not necessarily limited to) AttributeError, EOFError, ImportError,
    and IndexError.
```

**Comment by erikbern** [comment]
right – seems like python 3.5 now raises an `AttributeError` instead of `pickle.PicklingError`

**Comment by erikbern** [comment]
As a workaround you can just change the exception class to `Exception`.

**Comment by nchammas** [comment]
Yep, I'll update the PR accordingly since that's OK with you.

**Comment by erikbern** [comment]
nice! will merge if it runs

**Comment by nchammas** [comment]
Looks like `nonhdfs`, which failed last time, passed.

Am going to add updates to the README and setup.py and then this will be ready to merge in.

**Comment by erikbern** [comment]
Ok, was going to merge, but I'll hold off. Let me know when you are ready :)

**Comment by nchammas** [comment]
:japanese_ogre: Good to go. :japanese_goblin:

**Comment by erikbern** [comment]
thanks!

File: test/worker_parallel_scheduling_test.py [hunk 1]
```diff
@@ -149,7 +149,7 @@ def test_raise_unpicklable_exception_in_complete(self, send):
             UnpicklableExceptionTask().complete()
         except Exception as e:
             ex = e
-        self.assertRaises(pickle.PicklingError, pickle.dumps, ex)
+        self.assertRaises(Exception, pickle.dumps, ex)
```
Comment (Tarrasch) [inline test/worker_parallel_scheduling_test.py]: I remember you understood this part the best @daveFNbuck, does this code change look sensible to you?

Comment (nchammas) [inline test/worker_parallel_scheduling_test.py]: Original comment explaining why this change was made [here](https://github.com/spotify/luigi/pull/1494#issuecomment-170199567).

Comment (daveFNbuck) [inline test/worker_parallel_scheduling_test.py]: It would have been better to do `self.assertRaises((pickle.PicklingError, AttributeError), pickle.dumps, ex)` to ensure that we weren't catching too broadly.

Comment (nchammas) [inline test/worker_parallel_scheduling_test.py]: Agreed. I can open a new PR to fix that if you want.

Comment (Tarrasch) [inline test/worker_parallel_scheduling_test.py]: @nchammas, would be nice :)
